<a href="https://colab.research.google.com/github/liviaspereira/desafio_inferencia/blob/main/DesafioInferencia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Lendo os Banco de dados**

In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
from matplotlib import pyplot as plt

df1 = pd.read_csv("https://raw.githubusercontent.com/liviaspereira/desafio_inferencia/main/DEMO_PHQ.csv")
df2 = pd.read_csv("https://raw.githubusercontent.com/liviaspereira/desafio_inferencia/main/PAG_HEI.csv")


Este método **".info"** *imprime* informações sobre um DataFrame incluindo o dtype de índice e dtypes de coluna, valores não nulos e uso de memória.

In [ ]:
df1.info()
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5334 entries, 0 to 5333
Data columns (total 15 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      5334 non-null   int64  
 1   DPQ010    4836 non-null   float64
 2   DPQ020    4836 non-null   float64
 3   DPQ030    4836 non-null   float64
 4   DPQ040    4836 non-null   float64
 5   DPQ050    4836 non-null   float64
 6   DPQ060    4836 non-null   float64
 7   DPQ070    4836 non-null   float64
 8   DPQ080    4836 non-null   float64
 9   DPQ090    4836 non-null   float64
 10  RIAGENDR  5334 non-null   int64  
 11  RIDAGEYR  5334 non-null   int64  
 12  RIDRETH1  5334 non-null   int64  
 13  DMDEDUC   5334 non-null   int64  
 14  INDFMINC  5287 non-null   float64
dtypes: float64(10), int64(5)
memory usage: 625.2 KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9424 entries, 0 to 9423
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------    

In [ ]:
df1

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0     31130     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
1     31131     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2     31132     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3     31134     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
4     31139     0.0     0.0     0.0     0.0     3.0     1.0     0.0     0.0   
...     ...     ...     ...     ...     ...     ...     ...     ...     ...   
5329  41466     0.0     0.0     1.0     1.0     0.0     0.0     0.0     0.0   
5330  41468     0.0     2.0     0.0     1.0     1.0     2.0     1.0     3.0   
5331  41469     1.0     0.0     0.0     1.0     0.0     0.0     0.0     0.0   
5332  41472     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
5333  41473     0.0     0.0     2.0     1.0     0.0     1.0     0.0     0.0   

      DPQ090  RIAGENDR  RIDAGEYR  RIDRETH1  DMDEDUC  INDFMINC  
0        NaN         2        85         3        4       4.0  
1        0.0         2        44         4        4      11.0  
2        0.0         1        70         3        5      11.0  
3        0.0         1        73         3        3      12.0  
4        0.0         2        18         2        3      11.0  
...      ...       ...       ...       ...      ...       ...  
5329     0.0         2        58         5        2       3.0  
5330     0.0         2        66         1        1       8.0  
5331     0.0         1        19         4        4       2.0  
5332     0.0         1        34         3        4       7.0  
5333     0.0         1        21         1        2       6.0  

[5334 rows x 15 columns]

In [ ]:
df2

SEQN     PAG_MINW  ADHERENCE  HEI2015C1_TOTALVEG  \
0     31128          NaN        NaN            3.083350   
1     31129   285.450346        2.0            1.697761   
2     31130     0.000000        1.0                 NaN   
3     31131   264.711316        2.0            3.115103   
4     31132  1353.348730        3.0            3.509081   
...     ...          ...        ...                 ...   
9419  41469   184.711316        2.0            2.555935   
9420  41471   951.478060        3.0            2.674192   
9421  41472    27.713626        1.0            2.457422   
9422  41473   685.912240        3.0            3.907340   
9423  41474     0.000000        1.0            1.238646   

      HEI2015C2_GREEN_AND_BEAN  HEI2015C3_TOTALFRUIT  HEI2015C4_WHOLEFRUIT  \
0                     0.000000              3.927008              5.000000   
1                     0.000000              3.731602              4.929186   
2                          NaN                   NaN                   NaN   
3                     0.000000              5.000000              1.320674   
4                     0.000000              5.000000              5.000000   
...                        ...                   ...                   ...   
9419                  0.000000              2.613733              0.000000   
9420                  3.769401              2.198817              0.517369   
9421                  2.930799              2.311917              0.000000   
9422                  5.000000              0.000000              0.000000   
9423                  4.610515              0.688137              0.240848   

      HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
0                 0.000000              5.000000           4.251329   
1                 0.527631              3.887809           4.693141   
2                      NaN                   NaN                NaN   
3                 0.000000              3.141062           4.071959   
4                 5.000000              4.448512           5.000000   
...                    ...                   ...                ...   
9419              0.000000              2.151846           5.000000   
9420              0.463168              4.662005           5.000000   
9421              0.933303              5.000000           5.000000   
9422              0.000000              5.000000           5.000000   
9423              5.000000              5.000000           4.304982   

      HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
0                    0.000000             2.487572           6.911882   
1                    0.928562             3.740758           6.146441   
2                         NaN                  NaN                NaN   
3                    2.052399             0.684985           2.297100   
4                    0.000000             2.050769           7.369489   
...                       ...                  ...                ...   
9419                 0.000000             9.344947           3.535211   
9420                 1.191796             6.145146           3.189620   
9421                 5.000000             2.090534           8.409844   
9422                 5.000000             0.000000           7.458058   
9423                 5.000000             2.162293           7.018993   

      HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
0                   10.000000         0.818029           6.408092   
1                    7.274979         8.006457           0.000000   
2                         NaN              NaN                NaN   
3                    9.512964         0.438428           3.919226   
4                    8.887880         8.186877           9.099088   
...                       ...              ...                ...   
9419                 4.326307         5.269780           8.195564   
9420                 3.766149         3.974279           7.758940   
9421      

Combinar os dois bancos de dados fornecidos, utilizando a variável SEQN como chave única. O banco de dados final deve conter 5334 observações dos adultos respondentes do NHANES.

In [ ]:
df3 = df1.merge(df2, on = 'SEQN', how = 'left')

In [ ]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5334 entries, 0 to 5333
Data columns (total 31 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   SEQN                      5334 non-null   int64  
 1   DPQ010                    4836 non-null   float64
 2   DPQ020                    4836 non-null   float64
 3   DPQ030                    4836 non-null   float64
 4   DPQ040                    4836 non-null   float64
 5   DPQ050                    4836 non-null   float64
 6   DPQ060                    4836 non-null   float64
 7   DPQ070                    4836 non-null   float64
 8   DPQ080                    4836 non-null   float64
 9   DPQ090                    4836 non-null   float64
 10  RIAGENDR                  5334 non-null   int64  
 11  RIDAGEYR                  5334 non-null   int64  
 12  RIDRETH1                  5334 non-null   int64  
 13  DMDEDUC                   5334 non-null   int64  
 14  INDFMINC

In [ ]:
df3.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0  31130     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
1  31131     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2  31132     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3  31134     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
4  31139     0.0     0.0     0.0     0.0     3.0     1.0     0.0     0.0   

   DPQ090  ...  HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
0     NaN  ...                   NaN                   NaN                NaN   
1     0.0  ...              0.000000              3.141062           4.071959   
2     0.0  ...              5.000000              4.448512           5.000000   
3     0.0  ...              2.217331              2.313900           5.000000   
4     0.0  ...              0.211544              0.000000           4.088849   

   HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
0                      NaN                  NaN                NaN   
1                 2.052399             0.684985           2.297100   
2                 0.000000             2.050769           7.369489   
3                 1.268648             2.958985           0.000000   
4                 4.221442            10.000000           2.845774   

   HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
0                      NaN              NaN                NaN   
1                 9.512964         0.438428           3.919226   
2                 8.887880         8.186877           9.099088   
3                 5.627488         5.863078           5.878686   
4                 0.000000        10.000000           9.658507   

   HEI2015_TOTAL_SCORE  
0                  NaN  
1            35.553899  
2            63.551696  
3            42.886826  
4            47.074393  

[5 rows x 31 columns]

Crie a variável phq9 - escore Patient Health Questionnaire-9 como a soma das variáveis DPQ010 a DPQ090

In [ ]:
  df3["phq9"] = df3[["DPQ010", 
                   "DPQ020", 
                   "DPQ030", 
                   "DPQ040", 
                   "DPQ050", 
                   "DPQ060", 
                   "DPQ070", 
                   "DPQ080", 
                   "DPQ090"]].sum(axis = 'columns', skipna = False)

  df3[["DPQ010", 
     "DPQ020", 
     "DPQ030", 
     "DPQ040",
     "DPQ050", 
     "DPQ060", 
     "DPQ070", 
     "DPQ080", 
     "DPQ090",
     "phq9"]].head()

DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  DPQ090  \
0     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
1     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
4     0.0     0.0     0.0     0.0     3.0     1.0     0.0     0.0     0.0   

   phq9  
0   NaN  
1   0.0  
2   0.0  
3   0.0  
4   4.0

Criar uma variável phq_grp - grupo de sintomas de depressão que assume 
0 (“sem sintomas”) se phq9 < 5, 
1 (“sintomas leves”) se 5 <= phq9 < 10, 
2 (“sintomas moderados”) se 10 <= phq9 < 15, 
3 (“sintomas moderadamente severos”) se 15 <= phq9 < 19 e
4 (“sintomas severos”) se phq9 >= 20

In [ ]:
conditions = [
  (df3['phq9'].isna()),
  (df3['phq9'] <= 5),
  (df3['phq9'] > 5) & (df3['phq9'] <= 9),
  (df3['phq9'] > 9) & (df3['phq9'] <= 14),
  (df3['phq9'] > 14) & (df3['phq9'] <= 19),
  (df3['phq9'] > 19)
    ]
values = [np.nan, 0, 1, 2, 3, 4]

df3["phq_grp"] = np.select(conditions, values)

df3["phq_grp2"] = df3["phq_grp"]


In [ ]:
df3

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0     31130     NaN     NaN     NaN     NaN     NaN     NaN     NaN     NaN   
1     31131     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2     31132     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3     31134     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
4     31139     0.0     0.0     0.0     0.0     3.0     1.0     0.0     0.0   
...     ...     ...     ...     ...     ...     ...     ...     ...     ...   
5329  41466     0.0     0.0     1.0     1.0     0.0     0.0     0.0     0.0   
5330  41468     0.0     2.0     0.0     1.0     1.0     2.0     1.0     3.0   
5331  41469     1.0     0.0     0.0     1.0     0.0     0.0     0.0     0.0   
5332  41472     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
5333  41473     0.0     0.0     2.0     1.0     0.0     1.0     0.0     0.0   

      DPQ090  ...  HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  \
0        NaN  ...                      NaN                  NaN   
1        0.0  ...                 2.052399             0.684985   
2        0.0  ...                 0.000000             2.050769   
3        0.0  ...                 1.268648             2.958985   
4        0.0  ...                 4.221442            10.000000   
...      ...  ...                      ...                  ...   
5329     0.0  ...                 4.672204             4.633477   
5330     0.0  ...                 2.344264             4.780282   
5331     0.0  ...                 0.000000             9.344947   
5332     0.0  ...                 5.000000             2.090534   
5333     0.0  ...                 5.000000             0.000000   

      HEI2015C10_SODIUM  HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  \
0                   NaN                      NaN              NaN   
1              2.297100                 9.512964         0.438428   
2              7.369489                 8.887880         8.186877   
3              0.000000                 5.627488         5.863078   
4              2.845774                 0.000000        10.000000   
...                 ...                      ...              ...   
5329           0.000000                 6.580006         8.393244   
5330           3.134216                 1.986137         5.850269   
5331           3.535211                 4.326307         5.269780   
5332           8.409844                 8.936854         3.063795   
5333           7.458058                10.000000         1.676912   

      HEI2015C13_ADDSUG  HEI2015_TOTAL_SCORE  phq9  phq_grp  phq_grp2  
0                   NaN                  NaN   NaN      NaN       NaN  
1              3.919226            35.553899   0.0      0.0       0.0  
2              9.099088            63.551696   0.0      0.0       0.0  
3              5.878686            42.886826   0.0      0.0       0.0  
4              9.658507            47.074393   4.0      0.0       0.0  
...                 ...                  ...   ...      ...       ...  
5329           9.360990            59.949915   2.0      0.0       0.0  
5330          10.000000            57.747869  10.0      2.0       2.0  
5331           8.195564            42.993323   2.0      0.0       0.0  
5332           0.000000            46.134467   0.0      0.0       0.0  
5333           2.751168            45.793478   4.0      0.0       0.0  

[5334 rows x 34 columns]

Avaliando o percentual de casos faltantes (missing):

In [ ]:
df3.isnull().mean().round(4)*100

SEQN                        0.00
DPQ010                      9.34
DPQ020                      9.34
DPQ030                      9.34
DPQ040                      9.34
DPQ050                      9.34
DPQ060                      9.34
DPQ070                      9.34
DPQ080                      9.34
DPQ090                      9.34
RIAGENDR                    0.00
RIDAGEYR                    0.00
RIDRETH1                    0.00
DMDEDUC                     0.00
INDFMINC                    0.88
PAG_MINW                    0.00
ADHERENCE                   0.00
HEI2015C1_TOTALVEG          5.14
HEI2015C2_GREEN_AND_BEAN    5.14
HEI2015C3_TOTALFRUIT        5.14
HEI2015C4_WHOLEFRUIT        5.14
HEI2015C5_WHOLEGRAIN        5.14
HEI2015C6_TOTALDAIRY        5.14
HEI2015C7_TOTPROT           5.14
HEI2015C8_SEAPLANT_PROT     5.14
HEI2015C9_FATTYACID         5.14
HEI2015C10_SODIUM           5.14
HEI2015C11_REFINEDGRAIN     5.14
HEI2015C12_SFAT             5.14
HEI2015C13_ADDSUG           5.14
HEI2015_TO

O percentual de missing é bem baixo, dessa forma, a principio não se torna preocupante 

In [ ]:
df3.drop(columns=['SEQN']).describe(percentiles = [.25, .5, .75, .95, .99]).round(2)

DPQ010   DPQ020   DPQ030   DPQ040   DPQ050   DPQ060   DPQ070   DPQ080  \
count  4836.00  4836.00  4836.00  4836.00  4836.00  4836.00  4836.00  4836.00   
mean      0.31     0.31     0.55     0.72     0.32     0.21     0.23     0.15   
std       0.76     0.71     0.93     0.91     0.74     0.62     0.64     0.59   
min       0.00     0.00     0.00     0.00     0.00     0.00     0.00     0.00   
25%       0.00     0.00     0.00     0.00     0.00     0.00     0.00     0.00   
50%       0.00     0.00     0.00     0.00     0.00     0.00     0.00     0.00   
75%       0.00     0.00     1.00     1.00     0.00     0.00     0.00     0.00   
95%       2.00     2.00     3.00     3.00     2.00     1.00     1.00     1.00   
99%       3.00     3.00     3.00     3.00     3.00     3.00     3.00     3.00   
max       9.00     9.00     9.00     9.00     9.00     9.00     9.00     9.00   

        DPQ090  RIAGENDR  ...  HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  \
count  4836.00   5334.00  ...                  5060.00              5060.00   
mean      0.05      1.52  ...                     2.59                 4.71   
std       0.33      0.50  ...                     2.17                 3.24   
min       0.00      1.00  ...                     0.00                 0.00   
25%       0.00      1.00  ...                     0.00                 2.02   
50%       0.00      2.00  ...                     2.59                 4.43   
75%       0.00      2.00  ...                     5.00                 7.27   
95%       0.00      2.00  ...                     5.00                10.00   
99%       1.00      2.00  ...                     5.00                10.00   
max       9.00      2.00  ...                     5.00                10.00   

       HEI2015C10_SODIUM  HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  \
count            5060.00                  5060.00          5060.00   
mean                4.76                     5.80             6.07   
std                 3.42                     3.50             3.18   
min                 0.00                     0.00             0.00   
25%                 1.59                     2.93             3.70   
50%                 4.83                     6.23             6.41   
75%                 7.65                     9.25             9.00   
95%                10.00                    10.00            10.00   
99%                10.00                    10.00            10.00   
max                10.00                    10.00            10.00   

       HEI2015C13_ADDSUG  HEI2015_TOTAL_SCORE     phq9  phq_grp  phq_grp2  
count            5060.00              5060.00  4836.00  4836.00   4836.00  
mean                6.22                49.97     2.85     0.27      0.27  
std                 3.33                12.32     4.02     0.69      0.69  
min                 0.00                16.06     0.00     0.00      0.00  
25%                 3.72                41.03     0.00     0.00      0.00  
50%                 6.86                49.51     1.00     0.00      0.00  
75%                 9.45                58.17     4.00     0.00      0.00  
95%                10.00                71.19    11.00     2.00      2.00  
99%                10.00                79.86    18.00     3.00      3.00  
max                10.00                89.53    42.00     4.00      4.00  

[10 rows x 33 columns]

Avaliando frequências das variáveis qualitativas:



In [ ]:
df3[[
"DPQ010",
"DPQ020",
"DPQ030",
"DPQ040",
"DPQ050",
"DPQ060",
"DPQ070",
"DPQ080",
"DPQ090"]].agg(['value_counts'])

DPQ010       DPQ020       DPQ030       DPQ040       DPQ050  \
    value_counts value_counts value_counts value_counts value_counts   
0.0         3850         3769         3152         2429         3813   
1.0          673          769         1071         1706          691   
2.0          180          179          286          370          160   
3.0          124          114          320          326          168   
9.0            9            5            7            5            4   

          DPQ060       DPQ070       DPQ080       DPQ090  
    value_counts value_counts value_counts value_counts  
0.0         4134         4070         4362         4667  
1.0          513          545          341          122  
2.0           95          118           79           32  
3.0           89           99           45           13  
9.0            5            4            9            2